libdnf/dnf-context.h

```diff
@@ -51,6 +51,22 @@ struct _DnfContextClass
 };
 
 /**
+ * DnfContextCleanFlags:
+ * @DNF_CONTEXT_CLEAN_EXPIRE_CACHE:            Clean the indicator file for cache directories' age
+ * @DNF_CONTEXT_CLEAN_PACKAGES:                Clean the packages section for cache
+ * @DNF_CONTEXT_CLEAN_METADATA:                Clean the metadata section for cache directories
+ * @DNF_CONTEXT_CLEAN_ALL:                     Clean out all of the cache directories
+ *
+ * The clean flags for cache directories cleaning.
+ **/
+typedef enum {
+        DNF_CONTEXT_CLEAN_EXPIRE_CACHE          = (1 << 0),
+        DNF_CONTEXT_CLEAN_PACKAGES              = (1 << 1),
+        DNF_CONTEXT_CLEAN_METADATA              = (1 << 2),
+        DNF_CONTEXT_CLEAN_ALL                   = (1 << 3),
```

Comment:
I was going to say this was:
`DNF_CONTEXT_CLEAN_ALL = (DNF_CONTEXT_CLEAN_PACKAGES | DNF_CONTEXT_CLEAN_METADATA),`
But reading the code I think one difference is we'll even delete caches for repos that don't have an active configuration file in `/etc/yum.repos.d`.  So it's fine as is.